library/src/main/java/com/novoda/downloadmanager/File.java

```diff
@@ -0,0 +1,137 @@
+package com.novoda.downloadmanager;
+
+public class File {
+
+    private final String networkAddress;
+    private final Optional<DownloadFileId> downloadFileId;
+    private final Optional<FileName> fileName;
+    private final Optional<String> relativePath;
+
+    File(String networkAddress, Optional<DownloadFileId> downloadFileId, Optional<FileName> fileName, Optional<String> relativePath) {
```

Comment:
Especially since this is related to files, downloads etc. I found this class name confusing. This is supposed to be used by users, isn't it? Or is it internal? 

Considering it is public API, it could be annoying to use this if the user also have `java.io.File` imported in their class.

Reply:
I agree. What naming would you suggest?

Reply:
Hmm, maybe `BatchFile`? It already has references to batch.
I was thinking `FileDescriptor` but that's also a `java.io` class. 😄

Reply:
fa0527e